torch/optim/sparse_adam.py

```diff
@@ -99,7 +99,7 @@ def make_sparse(values):
                 if amsgrad:
                     torch.max(old_max_exp_avg_sq_values, exp_avg_sq_update_values, out=old_max_exp_avg_sq_values)
                     denom = old_max_exp_avg_sq_values.sqrt_().add_(group['eps'])
-                    max_exp_avg_sq = make_sparse(old_max_exp_avg_sq_values)
+                    max_exp_avg_sq.put_(grad_indices, old_max_exp_avg_sq_values)
```

Comment:
`old_max_exp_avg_sq_values` have been `sqrt`-ed in-place line above. This doesn't seem correct. The whole point of these `del`s was to prevent this kind of errors (trying to use squared averages after taking their root).

Reply:
@apaszke good catch! I changed the order of the two lines...